dialog/file.go

```diff
@@ -208,13 +279,27 @@ func (f *fileDialog) refreshDir(dir string) {
 	f.fileScroll.Refresh()
 }
 
-func (f *fileDialog) setDirectory(dir string) {
+func (f *fileDialog) setDirectory(dir fyne.ListableURI) error {
+	if dir == nil {
+		return fmt.Errorf("failed to open nil directory")
+	}
+
 	f.setSelected(nil)
 	f.dir = dir
 
 	f.breadcrumb.Children = nil
-	buildDir := filepath.VolumeName(dir)
-	for i, d := range strings.Split(dir, string(filepath.Separator)) {
+
+	if dir.Scheme() != "file" {
```

Comment:
Is this required? it seems like the dialog is abstracted now...

Reply:
I removed the check. ```StartingLocation``` is ignored if it doesn't have ```file``` as scheme.

Reply:
Why is it ignored? can we not allow nfs:// or ftp:// URIs in the directory picker?

Reply:
> can we not allow nfs:// or ftp:// URIs in the directory picker?

I don't know. That's just the way @charlesdaniels implemented it.

Reply:
I wrote it that way since we don't actually have support for those things yet, and at the time I was under the impression that it would just cause it to break later. That might not be the case anymore, so I think we can remove this check if others are under the impression it won't break.

I don't remember specifically setting it to ignore `StartingLocation` with non-`file` schemes. That was probably an oversight on my part, or again I was assuming it would break later and wanted to catch the error early. I should have commented it better.

Reply:
In ```effectiveStartingDir()``` the scheme gets checked against ```file``` and if it doesn't match the function just skips to ```PWD``` and ```HomeDir```.

Reply:
I don't think we should be enforcing the scheme type - if the StartingLocation is a valid ListableURI then we should use it.